Mathlib/Topology/Algebra/InfiniteSum/UniformOn.lean

```diff
@@ -0,0 +1,172 @@
+/-
+Copyright (c) 2025 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck, David Loeffler
+-/
+import Mathlib.Topology.Algebra.InfiniteSum.Defs
+import Mathlib.Topology.Algebra.UniformConvergence
+
+/-!
+# Infinite sum and products that converge uniformly on a set
+
+This file defines the notion of uniform convergence of infinite sums and products of functions,
+on a given family of subsets of their domain.
+
+It also defines the notion of local uniform convergence of infinite sums and products of functions
+on a set.
+-/
+
+noncomputable section
+
+open Filter Function
+
+open scoped Topology
+
+variable {α β ι : Type*} [CommMonoid α]  {f : ι → β → α} {g : β → α} {𝔖 : Set (Set β)}
+  {x : β} {s : Set β} {I : Finset ι} [UniformSpace α]
+
+/-!
+## Uniform convergence of sums and products
+-/
+
+section UniformlyOn
+
+variable (f g 𝔖)
+
+/-- `HasProdUniformlyOn f g 𝔖` means that the (potentially infinite) product `∏' i, f i b`
+for `b : β` converges uniformly on each `s` in a family of sets `𝔖` to `g`. -/
+@[to_additive "`HasSumUniformlyOn f g 𝔖` means that the (potentially infinite) sum `∑' i, f i b`
+for `b : β` converges uniformly on `s ∈ 𝔖` to `g`."]
+def HasProdUniformlyOn : Prop :=
+  HasProd (fun i ↦ UniformOnFun.ofFun 𝔖 (f i)) (UniformOnFun.ofFun 𝔖 g)
+
+/-- `MultipliableUniformlyOn f 𝔖` means that `f` converges uniformly on `s` to some infinite
+product. -/
+@[to_additive "`SummableUniformlyOn f s` means that `f` converges uniformly on `s` to some
+infinite sum."]
+def MultipliableUniformlyOn (f : ι → β → α) (𝔖 : Set (Set β)) : Prop :=
+  Multipliable (fun i ↦ UniformOnFun.ofFun 𝔖 (f i))
+
+variable {f g 𝔖}
+
+@[to_additive]
+lemma MultipliableUniformlyOn.exists (h : MultipliableUniformlyOn f 𝔖) :
+    ∃ g, HasProdUniformlyOn f g 𝔖 :=
+  h
+
+@[to_additive]
+theorem HasProdUniformlyOn.multipliableUniformlyOn (h : HasProdUniformlyOn f g 𝔖) :
+    MultipliableUniformlyOn f 𝔖 :=
+  ⟨g, h⟩
+
+@[to_additive]
+lemma hasProdUniformlyOn_iff_tendstoUniformlyOn : HasProdUniformlyOn f g 𝔖 ↔
+    ∀ s ∈ 𝔖, TendstoUniformlyOn (fun I b ↦ ∏ i ∈ I, f i b) g atTop s := by
+  simpa [HasProdUniformlyOn, HasProd, UniformOnFun.ofFun_prod, Finset.prod_fn] using
+    UniformOnFun.tendsto_iff_tendstoUniformlyOn
+
+@[to_additive]
+theorem HasProdUniformlyOn.hasProd (h : HasProdUniformlyOn f g 𝔖) (hs : s ∈ 𝔖) (hx : x ∈ s) :
+    HasProd (f · x) (g x) :=
+  (hasProdUniformlyOn_iff_tendstoUniformlyOn.mp h s hs).tendsto_at hx
+
+@[to_additive]
+theorem HasProdUniformlyOn.tprod_eqOn [T2Space α] (h : HasProdUniformlyOn f g 𝔖) (hs : s ∈ 𝔖) :
+    s.EqOn (∏' b, f b ·) g :=
+  fun _ hx ↦ (h.hasProd hs hx).tprod_eq
```

Comment:
Can you please provide a version that says `(∏' b, f b ·) = g` under the assumption `⋃₀ 𝔖 = Set.univ`?